twill-core/src/main/java/org/apache/twill/internal/kafka/client/SimpleKafkaConsumer.java

```diff
@@ -273,17 +274,19 @@ private MessageCallback wrapCallback(final MessageCallback callback,
       final AtomicBoolean stopped = new AtomicBoolean();
       return new MessageCallback() {
         @Override
-        public void onReceived(final Iterator<FetchedMessage> messages) {
+        public long onReceived(final Iterator<FetchedMessage> messages) {
           if (stopped.get()) {
-            return;
+            return Long.MIN_VALUE;
```

Comment:
Using AtomicLong as a mutable parameter is not good. In general, immutability gives cleaner API